File: paley.py
import turtle
import math
import sys

class Paley:

    def __init__(self, p, radius = 290, speed = 0):
        self.p = p
        self.radius = radius
        self.speed = speed

    """Return coordinates of ith vertex"""
    def getVertex(self, i):
        angle = i * 2 * math.pi / self.p
        return (self.radius * math.cos(angle), self.radius * math.sin(angle))

    """Draw the Paley graph"""
    def draw(self):
        t = turtle.Turtle()
        t.speed(self.speed)
        t.ht()
        for i in xrange(self.p):
            t.pu()
            t.goto(self.getVertex(i))
            for residue in xrange(1, (self.p - 1) / 2 + 1):
                j = (i + residue * residue) % self.p
                # Only draw the "forward" edges so each edge is only drawn once
                if j > i:
                    t.pd()
                    t.goto(self.getVertex(j))
                    t.pu()
                    t.goto(self.getVertex(i))
        turtle.done()

def main(): 
    if len(sys.argv) > 1:
        pal = Paley(int(sys.argv[1]))
        if len(sys.argv) > 2:
            pal.radius = int(sys.argv[2])
        pal.draw()
    else:
        print "Please specify size of Paley graph"

if __name__ == "__main__":
    main()

File: paley.js
// Return a new Paley drawer object with the following fields:
// -- p: a prime number === 1 mod 4
// -- canvas: an HTML 5 canvas object
// -- draw(): function to draw the Paley graph of order p on the canvas
// -- interval: handle for interval at which the edges of the graph are being drawn (so it can be passed to clearInterval()
var newPaley = function(p, canvas, speed) {
	var paley = {};
	paley.p = p;
	paley.canvas = canvas;

	// Smaller number == faster speed
	paley.speed = speed || 0;

	var getVertex = function(pal, i) {
		// These are also the coordinates of the center of the canvas
		var xRad = pal.canvas.width / 2;
		var yRad = pal.canvas.height / 2;
		var angle = i * 2 * Math.PI / pal.p;
		return {x: xRad + xRad * Math.cos(angle), y: yRad + yRad * Math.sin(angle)};
	};

	// Callback for setInterval
	// pal is a pointer to paley
	// params is an object containing the iteration variables i, j
	var drawLine = function(pal, params) {
	    if(params.i >= pal.p) {
			// Done
			clearInterval(pal.interval);
			return;
		}

		// Increment i and j until an edge has been drawn
		for(; params.i < pal.p; params.i++) {

			for(; params.j <= (pal.p - 1) / 2; params.j++) {

				var j = (params.i + params.j * params.j) % pal.p;

				// Only draw "forward" edges
				if(j > params.i) {
					var i = getVertex(pal, params.i);
					j = getVertex(pal, j);
					var ctx = pal.canvas.getContext("2d");
					ctx.beginPath();
					ctx.moveTo(i.x, i.y);
					ctx.lineTo(j.x, j.y);
					ctx.stroke();
					// Edge has been drawn; increment j and return
					params.j++;
					return;
				}	
			}
			params.j = 1;
		}
	};

	paley.draw = function() {
		// Object to hold iteration variables
		var indices = {i: 0, j: 1};
		this.interval = setInterval(drawLine, this.speed, this, indices);
	};

	return paley;
};
